scripts/generate-global-changelog.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const {execSync} = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const IGNORE_LIST = [
+	'packages/demos/CHANGELOG.md',
+	'clayui.com/CHANGELOG.md',
+	'CHANGELOG.md',
+];
+
+const dateObj = new Date();
+const month = dateObj.getMonth() + 1;
+const day = dateObj.getDate();
+const year = dateObj.getFullYear();
+
+const dateString = `${year}-${month}-${day}`;
+
+const globalChangeLog = path.join(__dirname, '../CHANGELOG.md');
+
+function main() {
+	const diffOutput = execSync(
+		'git diff HEAD^ --name-only -- "*CHANGELOG.md"'
+	).toString();
+
+	const changelogContents = fs.readFileSync(globalChangeLog).toString();
+
+	if (changelogContents.includes(`## ${dateString}`)) {
+		console.error(
+			`CHANGELOG has already been generated for the ${dateString} release`
+		);
+
+		return;
+	}
+
+	const files = diffOutput
+		.split(/\n/)
+		.filter((file) => !IGNORE_LIST.includes(file))
+		.filter(Boolean);
+
+	let output = `# CHANGELOG\n## ${dateString}\n`;
+
+	for (let i = 0; i < files.length; i++) {
+		const file = files[i];
+
+		try {
+			const fileDiff = execSync(`git diff HEAD^ -U0 -- "${file}"`)
+				.toString()
+				.split(/\n@@.*/)[1]
+				.trim()
+				.replace(/(\n|^)\+/g, '\n');
+
+			const packageName = file
+				.replace('packages/', '')
+				.replace('/CHANGELOG.md', '')
+				.replace('clay-', '@clayui/');
+
+			const pkgJson = file.replace('CHANGELOG.md', 'package.json');
+
+			const {version} = JSON.parse(fs.readFileSync(pkgJson));
+
+			const regexVersion = version.replace('.', '\\.');
+
+			const versionLinkRegex = `\\[${regexVersion}\\].*${regexVersion}\\)`;
+
+			const versionLinkMatch = fileDiff.match(
+				new RegExp(versionLinkRegex)
+			);
+
+			const versionLink = versionLinkMatch ? versionLinkMatch[0] : null;
+
+			// Be sure to properly nest headers in proper order.
+			// This order is CHANGELOG => date => package => fix type => commit
+			let content = fileDiff.replace(/(\n|^)#\s/gm, '\n#### ');
+			content = content.replace(/(\n|^)##\s/gm, '\n##### ');
+			content = content.replace(/(\n|^)###\s/gm, '\n###### ');
```

Comment:
If your goal is just to add `###` to each heading, I think you could do these in a single step:

```js
let content = fileDiff.replace(/^(#+)/gm, '$1###');
```

Note that because of the `m` flag, `^` will match at the beginning of any line, so you don't need to explicitly deal with `\n` anywhere.

If you want to keep the separate `replace()` calls you could chain them like you did on lines 58-60 above, for consistency.